src/providers/ipa/ipa_s2n_exop.c

```diff
@@ -2512,6 +2514,19 @@ static errno_t ipa_s2n_save_objects(struct sss_domain_info *dom,
                 }
             }
 
+            if (override_attrs != NULL) {
+                ret = sysdb_attrs_get_el_ext(override_attrs,
+                                             SYSDB_SSH_PUBKEY, false, &el);
+                if (ret == ENOENT) {
+                    ret = sysdb_set_user_attr(dom, name, &ssh_public_key_attrs,
+                                              SYSDB_MOD_DEL);
+                    if (ret != EOK) {
+                        DEBUG(SSSDBG_OP_FAILURE, "sysdb_set_user_attr failed,\n");
+                        goto done;
+                    }
+                }
+            }
+
```

Comment:
There is already a way to remove attributes from a user object in ipa_s2n_save_objects() with the help of the missing array. Currently it only supports one attribute, but I had to extend this for PR #488 in https://github.com/SSSD/sssd/pull/488/commits/4cee7b8e580d1e48cfc85c1ae146894e25a0262a. Feel free to copy the scheme here, I will then rebase my patches on top of yours.